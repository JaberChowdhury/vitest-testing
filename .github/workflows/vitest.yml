name: Run Vitest Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      # Step 1: Check out the repository
      - name: Checkout code
        uses: actions/checkout@v3

      # Step 2: Set up Node.js with pnpm caching
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: "pnpm"

      # Step 3: Install pnpm manually (bypassing Corepack)
      - name: Install pnpm
        run: npm install -g pnpm

      # Step 4: Ensure pnpm is added to PATH
      - name: Add pnpm to PATH
        run: echo "$(npm root -g)/pnpm" >> $GITHUB_PATH

      # Step 5: Verify pnpm installation and check PATH
      - name: Verify pnpm installation
        run: |
          echo "pnpm version:"
          pnpm --version
          echo "pnpm location:"
          which pnpm
          echo "Current PATH:"
          echo $PATH

      # Step 6: Install dependencies
      - name: Install dependencies
        run: pnpm install

      # Step 7: Run Vitest tests
      - name: Run tests
        run: pnpm test
