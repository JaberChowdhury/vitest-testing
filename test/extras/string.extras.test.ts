import { describe, test, expect } from "vitest";

describe("string matches using regex", () => {
  //  Regex Matching
  test("string matches regex", () => {
    const email = "user@example.com";
    expect(email).toMatch(/^[^\s@]+@[^\s@]+\.[^\s@]+$/); // Basic email regex
  });
  //  String Length
  test("string has correct length", () => {
    const str = "Vitest";
    expect(str).toHaveLength(6);
  });
  //  Custom String Assertions
  test("custom string assertion", () => {
    const str = "Vitest is awesome!";
    expect(str.startsWith("Vitest")).toBeTruthy();
    expect(str.endsWith("awesome!")).toBeTruthy();
  });

  //   Snapshot Testing for Strings
  // If you want to ensure that a string doesnâ€™t change unexpectedly, you can use snapshot testing.
  test("string snapshot", () => {
    const str = "This is a snapshot test and it shouldn't be change";
    expect(str).toMatchSnapshot();
  });
});
