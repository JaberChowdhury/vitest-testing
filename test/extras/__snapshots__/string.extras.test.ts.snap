// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`string matches using regex > string snapshot 1`] = `"This is a snapshot test and it shouldn't be change"`;
